ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp

```diff
@@ -0,0 +1,177 @@
+
+                    for (size_t ax = 0; ax < scales.size(); ++ax)
+                    {
+                        attrs.axes.insert(ax);
+                    }
+
+                    if (data_shape.is_static())
+                    {
+                        auto data_static_shape = data_shape.to_shape();
+
+                        std::vector<int64_t> output_shape;
+                        for (size_t i = 0; i < data_static_shape.size(); ++i)
+                        {
+                            output_shape.push_back(
+                                std::floor(data_static_shape.at(i) * scales.at(i)));
+                        }
+                        auto output_shape_const = default_opset::Constant::create(
+                            element::u64, Shape({output_shape.size()}), output_shape);
+
+                        return {std::make_shared<default_opset::Interpolate>(
+                            data, output_shape_const, attrs)};
+                    }
+
+                    const auto scales_const = default_opset::Constant::create(
+                        ngraph::element::f32, Shape({scales.size()}), scales);
+
+                    auto shape_of_data = std::make_shared<default_opset::Convert>(
+                        std::make_shared<default_opset::ShapeOf>(data), ngraph::element::f32);
+                    auto multiply =
+                        std::make_shared<default_opset::Multiply>(shape_of_data, scales_const);
+                    auto output_shape = std::make_shared<default_opset::Convert>(
+                        std::make_shared<default_opset::Floor>(multiply), ngraph::element::i64);
+                    return {
+                        std::make_shared<default_opset::Interpolate>(data, output_shape, attrs)};
+                }
+
+            } // namespace set_1
+
+            namespace set_9
+            {
+                NodeVector upsample(const onnx_import::Node& node)
+                {
+                    const auto inputs = node.get_ng_inputs();
+                    const auto data = inputs.at(0);
+                    const auto scales = inputs.at(1);
+
+                    const auto data_shape = data->get_output_partial_shape(0);
+                    const auto scales_shape = scales->get_output_partial_shape(0);
+
+                    const auto mode = node.get_attribute_value<std::string>("mode", "nearest");
+                    check_mode_support(node, mode);
+
+                    CHECK_VALID_NODE(
+                        node,
+                        (scales_shape.is_static() || data_shape.rank().is_static()),
+                        " Data rank or shape of Scales input is required to be static.");
+
+                    auto attrs = ngraph::op::v0::InterpolateAttrs();
+                    attrs.mode = mode;
+                    attrs.align_corners = false;
+
+                    size_t axes_size = scales_shape.is_static() ? scales_shape.to_shape().at(0)
```

Comment:
"_The number of elements of 'scales' should be the same as the rank of input 'X'._" I think you should enforce this condition.